Drivers/Src/Lights.c

```diff
@@ -9,16 +9,59 @@
 */
 
 
-
 #include "Lights.h"
+#include "common.h"
+#include <unistd.h>
+#include "BSP_SPI.h"
+#include "stm32f4xx.h"
+#include "Tasks.h"
+
+//static State lightStates[10] = {OFF};
+static uint16_t lightStates = 0x0000;
+static OS_MUTEX lightMutex;
+static CPU_TS timestamp;
+static OS_ERR err;
 
 /**
 * @brief   Initialize Lights Module
 * @return  void
 */ 
 void Lights_Init(void) {
+    OSMutexCreate(&lightMutex, "Light lock", &err); //init mutex
+    if(err != OS_ERR_NONE){
+        assertOSError(OS_BLINK_LIGHTS_LOC, err);
+    }
+
+    OSMutexPend(&lightMutex,0,OS_OPT_PEND_BLOCKING,&timestamp,&err); //lock light hardware for initialization (TODO: may not be needed, just here for additional safety)
+    if(err != OS_ERR_NONE){
+        assertOSError(OS_BLINK_LIGHTS_LOC, err);
+    }
+
     BSP_GPIO_Init(LIGHTS_PORT, 0x3C0, 1); // Pins 6,7,8,9 from Port C are out (0b1111000000)
     BSP_SPI_Init();
+
+    uint8_t txReadBuf[2] = {SPI_OPCODE_R, 0x01}; //0x01 is IODIRB in Bank 0 Mode
```

Comment:
It might be nice if there was an `enum` or a `#define` for this instead of just `0x01`